src/form/iv.jl

```diff
@@ -1,126 +1,68 @@
-    variable_voltage_real(pm; nw=nw, cnd=cnd, bounded=bounded, kwargs...)
-    variable_voltage_imaginary(pm; nw=nw, cnd=cnd, bounded=bounded, kwargs...)
-end
 
 ""
 function variable_branch_current(pm::AbstractIVRModel; kwargs...)
     variable_branch_current_rectangular(pm; kwargs...)
+    variable_branch_series_current_rectangular(pm; kwargs...)
 end
 
 ""
-function variable_load(pm::AbstractIVRModel; kwargs...)
-    variable_load_current_rectangular(pm; kwargs...)
+function variable_gen(pm::AbstractIVRModel; nw::Int=pm.cnw, cnd::Int=pm.ccnd, bounded::Bool=true, kwargs...)
+    variable_gen_current_rectangular(pm, cnd=cnd, nw=nw, bounded=bounded; kwargs...)
+    if bounded
+        for (i,gen) in ref(pm, nw, :gen)
+            constraint_gen_active_power_limits(pm, i, cnd=cnd, nw=nw)
+            constraint_gen_reactive_power_limits(pm, i, cnd=cnd, nw=nw)
+        end
+    end
 end
 
 ""
-function variable_gen(pm::AbstractIVRModel; kwargs...)
-    variable_gen_current_rectangular(pm; kwargs...)
+function variable_dcline(pm::AbstractIVRModel; nw::Int=pm.cnw, cnd::Int=pm.ccnd, bounded::Bool=true, kwargs...)
+    variable_dcline_current_rectangular(pm, cnd=cnd, nw=nw, bounded=bounded; kwargs...)
+    if bounded
+        for (i,dcline) in ref(pm, nw, :dcline)
+            constraint_dcline_power_limits_from(pm, i, cnd=cnd, nw=nw)
+            constraint_dcline_power_limits_to(pm, i, cnd=cnd, nw=nw)
+        end
+    end
 end
 
-""
-function variable_dcline(pm::AbstractIVRModel; kwargs...)
-    variable_dcline_current_rectangular(pm; kwargs...)
-end
+"""
+Defines how current distributes over series and shunt impedances of a pi-model branch
+"""
+function constraint_current_from(pm::AbstractIVRModel, n::Int, c::Int, f_bus, f_idx, g_sh_fr, b_sh_fr, tr, ti, tm)
+    vrfr = var(pm, n, c, :vr, f_bus)
+    vifr = var(pm, n, c, :vi, f_bus)
 
+    csrfr =  var(pm, n, c, :csr, f_idx)
+    csifr =  var(pm, n, c, :csi, f_idx)
 
+    crfr =  var(pm, n, c, :cr, f_idx)
+    cifr =  var(pm, n, c, :ci, f_idx)
 
-"`v[i] == vm`"
-function constraint_voltage_magnitude_setpoint(pm::AbstractIVRModel, n::Int, c::Int, i, vm)
-    vr = var(pm, n, c, :vr, i)
-    vi = var(pm, n, c, :vi, i)
-
-    JuMP.@constraint(pm.model, (vr^2 + vi^2) == vm^2)
+    JuMP.@constraint(pm.model, crfr == (tr*csrfr - ti*csifr + g_sh_fr*vrfr - b_sh_fr*vifr)/tm^2)
+    JuMP.@constraint(pm.model, cifr == (tr*csifr + ti*csrfr + g_sh_fr*vifr + b_sh_fr*vrfr)/tm^2)
 end
 
-"`vmin <= vm[i] <= vmax`"
-function constraint_voltage_magnitude(pm::AbstractIVRModel, n::Int, c::Int, i, vmin, vmax)
-    @assert vmin <= vmax
-    vr = var(pm, n, c, :vr, i)
-    vi = var(pm, n, c, :vi, i)
+"""
+Defines how current distributes over series and shunt impedances of a pi-model branch
+"""
+function constraint_current_to(pm::AbstractIVRModel, n::Int, c::Int, t_bus, f_idx, t_idx, g_sh_to, b_sh_to)
+    vrto = var(pm, n, c, :vr, t_bus)
+    vito = var(pm, n, c, :vi, t_bus)
```

Comment:
nit, `vr_to` and `vi_to` would be more consistent with the rest of the codebase.

Reply:
ok